### ISSUE TITLE
[Master Feature] Add GDPR support for AMP Pages

### ISSUE THREAD

**Issue by jasti** [ampproject/amphtml#12968]
Please note that the requirements are still in draft and subject to change as the law guidance becomes more concrete. At the minimum, the following functionality is necessary to build in AMP:
1. Ability to set a flag by the publisher to not send any 3rd party cookies and client id as part of the ad server request until the user has accepted a consent
2. Ability for a publisher page to only make an ad request only once the user has explicitly accepted amp-user-notification

CC @rudygalfi @lannka

**Comment by lannka** [comment]
@zhouyx

**Comment by Malvoz** [comment]
@jasti, typo.. that would be **GDPR**. Does the GDPR cookie consent apply to amp-iframe aswell?

**Comment by jasti** [comment]
Updated, thanks.
@rudygalfi for comments on amp-iframe.

**Comment by rudygalfi** [comment]
@Malvoz Could you clarify your question? Are you asking whether a cookie consent (these are usually implemented in AMP using `amp-user-notification`) can be applied to an `amp-iframe`?

**Comment by Malvoz** [comment]
@rudygalfi 

Yes I initially meant the latter - if cookie consent should be applied in general when there is an iframe with a source that is cross origin. If a publisher set a flag to not send **any** third party cookies, I thought it should apply to iframe content aswell.

If this feature wont include iframes by default, can pulishers still apply cookie consent using `amp-user-notification`? If not, publishers could turn to sandboxing iframes.

**Comment by rudygalfi** [comment]
@Malvoz: Please see further discussion in #13716 around offering support to manage collection of consent. We are also thinking through how various AMP elements could be configured to behave (e.g. send a non-personalized request vs. a personalized one) and will include details soon. cc @zhouyx

**Comment by jasti** [comment]
Closing in favor of #13716

**Comment by src-code** [comment]
[I left a comment over on the I2I issue earlier](https://github.com/ampproject/amphtml/issues/13716#issuecomment-378485717), but since this issue's just been reopened I'll rephrase my question here since it's specific to GDPR and AMP.

My question is, whose responsibility is it to obtain GDPR consent for cached AMP pages?  Is it the publisher of the content, or the site who serves the AMP page to the user (eg Google Search?)  My belief (and the sense I get from talking to lawyers) is it's the latter, since a cached AMP page is no different than other types of embedded content that may beacon to a 3rd party domain (eg ads).  

If that's the case, then should publishers omit GDPR consent flows (eg `amp-consent`) in AMP markup that's served to bots?  

In the Google Search example, Google is serving AMP content to its users from a Google-owned domain (cdn.ampproject.org), and that content is usually sending data to a third parties (the publisher domain or other analytics providers.)  It's not clear that the publisher has any obligation to obtain the consent in this scenario, since the user is Google's and only visiting Google domains.  

Contrast this with users who bypass the cache and visit an AMP page directly, eg Twitter users who are navigated directly to the publisher's AMP endpoint through a URL rewrite.  In this case, it'd make sense for the publisher to obtain GDPR consent since the user is visiting their domain directly.  Same for publishers who are serving AMP as their primary content experience.

So should publishers serve different content (sans `amp-consent`) to AMP cache bots given this difference in responsibility for consent?  Or is it Google's position that cached AMP is no different from uncached AMP under GDPR?  I can't imagine that Google would want its users having to click through consent flows for every publisher that appears in the AMP carousel, especially if it's unnecessary from a legal perspective.  

Clarity around this would be very helpful as we plan how to use these new consent tools in our AMP pages!

**Comment by rudygalfi** [comment]
@src-code Please see reply over at https://github.com/ampproject/amphtml/issues/13716#issuecomment-379814502

**Comment by jasti** [comment]
Closing this again in favor of #13716
